src/marshmallow_annotations/ext/dataclass.py

```diff
@@ -33,9 +35,14 @@ def _has_default(field: Field) -> bool:
 
 
 def _initvar_converter(
-    converter: AbstractConverter, subtype: Tuple[type], opts: ConfigOptions
+    converter: BaseConverter, subtype: Tuple[type], opts: ConfigOptions
 ) -> FieldABC:
-    return converter.convert(subtype[0], opts)
+    if 'initvar-typehint' not in opts.get("metadata", {}):
+        raise ValueError("use field(metadata=initvar_typehint(<typehint)) "
+                         "or field(..., metadata={**other_metadata, **initvar_typehint(<typehint>)})")
+    real_typehint = opts["metadata"]["initvar-typehint"]
+    real_field_constructor = converter.registry.get(real_typehint)
```

Comment:
Bit ugly, but functional.

P.S. AbstractConverter has no `registry` attribute, which we need. Wasn't sure whether adding to to AbstractConverter would be appropriate, so updated this fn to specify that` BaseConverter` is required

Reply:
It does involve registering the type twice (the second time with marsh_annotation specific stuff). Definitely not wonderful, but here's my reasoning for why it should be considered:

If you already have dataclasses lying around (which include init-vars) and you want to start using marshmallow on them then:
1.) this requires no observable changes to existing classes (no library should change behavior based on presence of keys/values in metadata besides their own keys/values)
2.) The minimum possible amount of duplication, short of being able to pull typing information out of the InitVar. If you want nested deserialization to work, e.g. `field_a: InitVar[OtherDataclassB]`, then *somewhere* you have to tell marshmallow-annotations that it should deserialize field_a using the registry before passing it into the `__init__` method
3.) The duplication exists on the same line (usually) and at least within the same declaration. This means if the two type hints get out of sync it's much easier to spot at a glance (in a PR or later) than if they're physically separated somehow.